CustomHooks.php

```diff
@@ -269,6 +272,49 @@ function _generate_new_wiki_username($info) {
         }
         return $username;
     }
+
+    # if Caliper is setup, use a custom actor with puid from LDAP
+    if (getenv('CaliperHost') && getenv('CaliperAPIKey')) {
+        $wgHooks['SetCaliperActorObject'][] = 'SetCaliperActor';
+
+        // This is the username MediaWiki will use.
+        function SetCaliperActor(&$actor, &$user) {
+            global $wgDBprefix;
+
+            if ($actor !== null) {
+                return true;
+            } else if (!$user->isLoggedIn() || !$user->getId()) {
+                return false;
+            }
+
+            $puid = null;
+            $userId = $user->getId();
+
+            $dbr = wfGetDB(DB_REPLICA);
+            $res = $dbr->select(
+                array('ucead' => $wgDBprefix.'user_cwl_extended_account_data'),   // tables
+                array('ucead.puid'),       // fields
```

Comment:
This seems UBC specific. Could you add some comments here or in readme? If other people want to use this, make sure pointing them to change this. or make this configurable.

Reply:
Added a readme section on it https://github.com/ubc/mediawiki-docker/pull/7/files#diff-04c6e90faac2675aa89e2176d2eec7d8R122

The LDAP hooks in the file are also fairly UBC specific so I think the best way to handle it would be to recommend others customize it to their needs (I actually make the `SetCaliperActorObject` Hook in the caliper extension just so that it would be easier to customize it to different needs like our own).

Reply:
👍